src/libcore/tests/result.rs

```diff
@@ -198,7 +198,7 @@ pub fn test_unwrap_or_default() {
 #[test]
 fn test_try() {
     fn try_result_some() -> Option<u8> {
-        let val = Ok(1)?;
+        let val = Ok::<_, NoneError>(1)?;
```

Comment:
Personally I'm not too bothered by this as `Ok(1)?` instead of `Some(1)?` is a silly thing to do; I don't know why `Ok(1)?` works in the first place...

Reply:
I guess inference can tell that the only possible `From` for `NoneError` is the blanket identity `impl<T> From<T> for T`, versus now inferring some `T: Into<NoneError>`.